src/code/bank0.asm

```diff
@@ -6702,19 +6702,27 @@ FillRoomWithConsecutiveObjects::
     ret                                           ;; 00:34FF $C9
 
 ; On GBC, special case for some overworld objects
-label_3500::
-    cp   $04                                      ;; 00:3500 $FE $04
+; Part of the macro objects system
+; Identical to SetupDestroyableObjectIfNeeded, but with different return banks 
+; Input:
+;   a    object id
+SetupDestroyableObjectIfNeeded2::
```

Comment:
I know this is a bad name, but I'm not sure what the difference between the two subroutines is. They set up different return banks, but I don't know why. It looks like this one might be used indoors and the other one outdoors, but that they simply copied it and left the unused outdoors-specific code in this one, like the Stone Pig Head etc.

Reply:
Two of the same functions with slight variations happens more in the code. I think the naming with numbers is fine until we better understand why they exist (which might be never understood)